### PR TITLE
feat: Add Configuration of TrustedEngines and add to Request Metadata

### PR DESCRIPTION
**Pull request by Midor**
## Summary by CodeRabbit

* **New Features**
  * Modular trusted engine configuration added, supporting Trino engines and selectable security models (Invoker/Definer).
  * Request metadata now records identified trusted engines during authentication.
  * Authentication middleware auto-detects and associates trusted engines based on token audiences and IdP; ambiguous matches now cause an unauthorized response.

* **Chores**
  * Switched limes dependency from crates.io version to a specific git revision.

### CODE REVIEW

**Comment by Midor** [comment]
@c-thiel In the tests for the environment variables I also added one version using dashes and one using underscores. I am not sure about the format of the audiences. So if there can be any edge cases where it would fail, because it cannot be represented as environment variable.

File: Cargo.toml [hunk 1]
```diff
@@ -100,7 +100,7 @@ jwks_client_rs = { version = "0.5.1", default-features = false, features = [
 ] }
 lazy-regex = { version = "3.2.0", features = ["lite"] }
 lazy_static = "^1.4"
-limes = { version = "0.3.1", features = ["kubernetes", "axum", "rustls-tls"] }
+limes = { git = "https://github.com/vakamo-labs/limes-rs.git", rev = "a88da41", features = ["kubernetes", "axum", "rustls-tls"] }
```
[the remark this answers is not in the document]
Comment (Midor) [reply Cargo.toml:103]: This is a temporary link to a specific commit, which in fact exists, as the tests worked. So we can resolve this for now. But yes, in the future we need to switch back to a fixed version from crates.io

**Comment by Midor** [comment]
@coderabbitai review

**Comment by Midor** [comment]
@coderabbitai review and resolve if fixed

**Comment by Midor** [comment]
@c-thiel FYI: This PR will also temporarily change the `limes` dependency to a specific commit (branch in `limes` with PR). After the PR in `limes` is merged **and** before we make a new release, we should change this back. But until then, I'll merge this commit.
